src/sentry/event_manager.py

```diff
@@ -2266,7 +2250,8 @@ def _calculate_event_grouping(
             event.data["grouping_config"] = get_grouping_config_dict_for_project(project)
             hashes = event.get_hashes()
 
-    hashes.write_to_event(event.data)
+    # Using cast to satisfy mypy
+    hashes.write_to_event(cast(Dict[str, Any], event.data))
```

Comment:
the types should get updated here rather than `cast` -- `cast` and `type: ignore` should generally only be used for mypy bugs